nimble/controller/src/ble_ll_scan.c

```diff
@@ -2086,8 +2100,10 @@ ble_ll_scan_parse_ext_hdr(struct os_mbuf *om,
     out_evt->sec_phy = aux_data->aux_phy;
     out_evt->prim_phy = aux_data->aux_primary_phy;
 
-    /* Adjust mbuf to contain advertising data only */
-    os_mbuf_adj(om, i);
+    if (i) {
```

Comment:
I would check `ext_hdr_len` here so it's easier to follow (there's unconditional `i = ext_hdr_len` few lines earlier)

Reply:
agree